tesseract/integration-test/src/lib.rs

```diff
@@ -0,0 +1,482 @@
+
+	while let Some(block_stream) = finalized_blocks_b_stream.next().await {
+		match block_stream {
+			Ok(block) => {
+				let state_machine_update = StateMachineUpdated {
+					state_machine_id: chain_a_client.state_machine_id(),
+					latest_height: (block.number() as u64) + 1,
+				};
+
+				if let Ok(events) = chain_a_client
+					.query_ismp_events(tx_block_height - 1, state_machine_update)
+					.await
+				{
+					if let Some(event) = events.into_iter().find_map(|event| match event {
+						Event::GetRequest(_) => Some(event),
+						_ => None,
+					}) {
+						get_request = Some(event.clone());
+						break
+					} else {
+						continue
+					}
+				} else {
+					continue
+				}
+			},
+			Err(err) => panic!("No next block in the stream {:?}", err),
+		}
+	}
+	// ======== process the request offchain ( 1. Make the response ) =============
+	let response = {
+		let get = match get_request.unwrap() {
+			Event::GetRequest(get) => get,
+			_ => panic!("Not supported"),
+		};
+		let dest_chain_block_hash = client_b.rpc().block_hash(Some(get.height.into())).await?;
+		let keys = get.keys.iter().map(|key| &key[..]).collect::<Vec<&[u8]>>();
+		let value_proof = client_b.rpc().read_proof(keys, dest_chain_block_hash).await?;
+		let proof = value_proof.proof.into_iter().map(|bytes| bytes.0).collect::<Vec<Vec<u8>>>();
+
+		let proof_of_value = SubstrateStateProof::StateProof(StateMachineProof {
+			hasher: HashAlgorithm::Keccak,
+			storage_proof: proof,
+		});
+		let proof = Proof {
+			height: StateMachineHeight {
+				id: chain_b_client.state_machine_id(),
+				height: get.height,
+			},
+			proof: proof_of_value.encode(),
+		};
+		let response = ResponseMessage {
+			datagram: RequestResponse::Request(vec![Request::Get(get.clone())]),
+			proof,
+			signer: chain_a_client.address(), // both A&B have same relayer address
+		};
+
+		Message::Response(response)
+	};
+	// =================== send to the source chain ================================
+	let _res = chain_a_client.submit(vec![response]).await?;
+	//==================== after approx 7-9 blocks the response event is emitted ===
+	// =================== fetch the returned value ================================
+
+	let mut response_event: Option<GetResponse> = None;
+
+	let mut finalized_blocks_stream = client_a.blocks().subscribe_finalized().await?;
+	while let Some(block) = finalized_blocks_stream.next().await {
+		match block {
+			Ok(block) => {
+				let block_hash = block.hash();
+				let fetched_events =
+					client_a.events().at(block_hash).await?.find_first::<GetResponse>()?;
+				match fetched_events {
+					Some(res_event) => {
+						response_event = Some(res_event.clone());
+						break
+					},
+					None => continue,
+				}
+			},
+			Err(err) => {
+				panic!("Error in finalized block stream: {:?}", err)
+			},
+		}
+	}
+
+	let encoded_value = match response_event.unwrap().0[0].clone() {
+		Some(value) => value,
+		None => {
+			panic!("Value not found")
+		},
+	};
```

Comment:
Unneeded, submit the transaction manually and wait for it to finalize and get its events